### PR TITLE
Rain did name this

### PR DESCRIPTION
**Pull request by Rainchus**

### CODE REVIEW

File: src/code_182B30.c [hunk 1]
```diff
@@ -175,7 +175,22 @@ INCLUDE_ASM(s32, "code_182B30", func_802597B0);
 
 INCLUDE_ASM(s32, "code_182B30", func_8025995C);
 
-INCLUDE_ASM(s32, "code_182B30", func_80259A48);
+void func_80259A48(s32 arg0, ActorPart* arg1, s32 arg2, s32 arg3) {
+    DecorationTable* decorationTable = arg1;
```
Comment (nanaian) [inline src/code_182B30.c:179]: is this assignment required?

File: src/code_182B30.c [hunk 2]
```diff
@@ -199,15 +214,57 @@ INCLUDE_ASM(s32, "code_182B30", func_8025BAA0);
 
 INCLUDE_ASM(s32, "code_182B30", func_8025C120);
 
-INCLUDE_ASM(s32, "code_182B30", func_8025C840);
+s32 func_8025C840(s32 arg0, ActorPart *arg1) {
 
-INCLUDE_ASM(s32, "code_182B30", func_8025C8A0);
+    if ((arg1->flags & 2) == 0) {
```
Comment (nanaian) [inline src/code_182B30.c:219]: use `!(arg1->flags & 2)` pls

File: src/code_182B30.c [hunk 3]
```diff
@@ -199,15 +214,57 @@ INCLUDE_ASM(s32, "code_182B30", func_8025BAA0);
...
+    }
+}
 
 INCLUDE_ASM(s32, "code_182B30", func_8025C918);
 
-INCLUDE_ASM(s32, "code_182B30", func_8025CCC8);
+s32 func_8025CCC8(s32 arg0, ActorPart *arg1) {
 
-INCLUDE_ASM(s32, "code_182B30", func_8025CD28);
+    if ((arg1->flags & 2) == 0) {
```
Comment (nanaian) [inline src/code_182B30.c:247]: Same here

**Review by nanaian — CHANGES_REQUESTED** [review]

File: include/common_structs.h [hunk 4]
```diff
@@ -1187,7 +1187,19 @@ typedef struct CollisionStatus {
 } CollisionStatus; // size = 0x28
 
 typedef struct DecorationTable {
-    /* 0x000 */ char unk_00[2011];
+    /* 0x000 */ char unk_00[1729];
```
Comment (ethteck) [inline include/common_structs.h:1190]: If an unk is being used by code, could you make the `char`s s8 instead? `char` in structs is used to denote unknown space

File: src/code_182B30.c [hunk 5]
```diff
@@ -175,7 +175,22 @@ INCLUDE_ASM(s32, "code_182B30", func_802597B0);
...
+    DecorationTable* decorationTable = arg1;
+
+    decorationTable = arg1->decorationTable;
+    if (decorationTable->unk_6C1 != 0) {
+        arg1->verticalStretch = 1;
+        arg1->unkOffset[0] = 0;
+        arg1->unkOffset[1] = 0;
+        decorationTable->unk_6C1 = 0;
+    }
+    if (arg0 == 0) {
```
Comment (ethteck) [inline src/code_182B30.c:188]: I think if/else-ing this would look nicer

**Review by ethteck — CHANGES_REQUESTED** [review]

File: src/code_182B30.c (hunk 5 above)
Comment (JoshDuMan) [reply src/code_182B30.c:188]: I don't think thats equivalent lol

Comment (JoshDuMan) [reply src/code_182B30.c:188]: OH nvm

File: src/code_182B30.c (hunk 1 above)
Comment (Rainchus) [reply src/code_182B30.c:179]: I was having issues without it

Comment (ethteck) [reply src/code_182B30.c:179]: You're assigning an ActorPart* to a DecorationTable* though. if you can't find a way to clean this up, could you leave a comment saying "TODO look into" or something?

**Review by ethteck — APPROVED** [review]

**Review by nanaian — APPROVED** [review]

**Pull request merged**